packages/react-components/src/components/chart/eChartsWrapper.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useEffect, useRef } from 'react';
+import { init, getInstanceByDom } from 'echarts';
+
+import type { ECharts, EChartsOption, SetOptionOpts } from 'echarts';
+
+export interface EChartsWrapperProps {
+  option: EChartsOption;
+  settings?: SetOptionOpts;
+  loading?: boolean;
+  theme?: string;
+  className?: string;
+  size?: { width?: number; height?: number };
+}
+
+export const EChartsWrapper = ({ option, settings, loading, theme, size = {} }: EChartsWrapperProps): JSX.Element => {
+  const chartRef = useRef<HTMLDivElement>(null);
+
+  // Initialize chart + listeners
+  useEffect(() => {
+    let chart: ECharts | undefined;
+    if (chartRef.current !== null) {
+      chart = init(chartRef.current, theme, { ...size });
+    }
+
+    return () => {
+      chart?.dispose();
+    };
+  }, []);
+
+  // Update chart if theme, options, or settings change
+  useEffect(() => {
+    if (chartRef.current !== null) {
+      const chart = getInstanceByDom(chartRef.current);
+      chart?.setOption(option, settings);
+    }
+  }, [chartRef, option, settings, theme]);
+
+  useEffect(() => {
+    if (chartRef.current !== null) {
+      const chart = getInstanceByDom(chartRef.current);
+      loading === true ? chart?.showLoading() : chart?.hideLoading();
```

Comment:
maybe we want a custom element, but I think echarts has some loading options - https://echarts.apache.org/en/api.html#echartsInstance.showLoading - have not messed around with it though